cmd/samples/recipes/ctxpropagation/propagator.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+
+	"go.uber.org/cadence/workflow"
+)
+
+type (
+	contextKey struct{}
+
+	propagator struct{}
```

Comment:
Can you also include some documentation on this type.  Like what is the purpose, which interface it implements, translation of header properties -> context, etc.